awx/main/models/base.py

```diff
@@ -29,6 +29,10 @@
 PERM_INVENTORY_CHECK  = 'check'
 PERM_INVENTORY_SCAN   = 'scan'
 
+JOB_EVENTS_ALL = "all"
+JOB_EVENTS_ONLY_ERRORS = "errors"
+JOB_EVENTS_NONE = "none"
```

Comment:
You have 3 options here.

The settings in Ansible core that do almost the same thing have 2 toggles.

https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/doc_fragments/default_callback.py#L11-L36.

I agree that we should have 1 more toggle than what they have - because verbose events are not a concept to Ansible core. Also, I think it's a reasonable feature suggestion for Ansible to add toggles for `display_changed_hosts` and `display_failed_hosts`. I can try to make a PR which does that if you're interested in.

In any scenario, I don't see a justification for creating a new kind of event-status-filter logic. It's not about the difficulty of doing it, users themselves have to learn to use it, we have to maintain it, and so on...

To put forward a proposal, let's say we introduce these new options to the Ansible runner display callback plugin.

 - record_verbose_events
 - record_failed_hosts
 - record_changed_hosts
 - record_ok_hosts
 - record_skipped_hosts

(necessary piece of technical background: our display callback inherits those options like display_ok_hosts)

Instead of _hosts maybe _events would be nicer.

We can easily add these to the callback plugin. Then it'll be picked up by the job template as the user puts it in their `ansible.cfg` in the project. This has the added benefit to make it configurable along with the display of events. Then if we start to surface these options in the UI at some point, it'll be a coherent unified interface, and not disjointed implementations that all seem like they were done ignorant of the others.

Reply:
@AlanCoding but if we go this route, we won't be able to record the "extra" events to the external logger (maybe @matburt doesn't care about that aspect as much as I thought he did?)

Reply:
I feel like you may be missing the whole point of this. The point is not to short circuit event creation entirely. We *definitely* want the events to flow through and come out of the Runner interface especially in order to be propagated down to the logger. We just *don't* want those events in the database.

It'd be a very short jump to filter these at the Ansible level or, better, the Runner level... but we *need* these events to come through and at least make it to the logger. *That* is the point of this PR.

Reply:
For what it's worth, there's certainly value in short circuiting things entirely and filtering at the Runner or Ansible level, that's just not what I'm going for here.

Reply:
This PR adds a toggle in the job template form in the UI, and if a user messes with that toggle, some event types stop showing up in the job standard out. Whether or not you intended this to be that, that becomes an implementation of #4092, and having first-class visibility to the user, it becomes the canonical implementation.

I'm sure it's fine for the problem you're trying to address, that's not my concern.

Now, for anyone to do a hard filter of event types (the way it probably should be done in most cases), they can put it in `ansible.cfg` in their project and have it work. It's just all backwards. What you have here is a very specific and technical function for a subset of users, but _that's_ what gets in the UI, and the Ansible settings are left to the obscure uses of Googling stuff, putting it in source control, and so on.

Reply:
Why not a toggle to do what you're doing here, but specifically for events that have empty standard out. Or better yet, give a range of options, (nothing), (log), or (record & log)

Users already have the ability to control what goes in standard out. They have the ability to print to standard out for the runner-on-start events. For other event types, they have the ability to silence them.

Today, that ability does not solve the problem they have because our system still records events even if the user turns off the display of them. The users who wanted this are more sophisticated than what I feel like this gives them credit for. If they wanted to hide the events, they can do it, but it won't reduce database use by much, because the server fights against them by saving it anyway. I'm all for fixing that without changing the default behavior.